src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationHandler.cs

```diff
@@ -24,7 +24,12 @@ public LinkedInAuthenticationHandler([NotNull] HttpClient client)
 
         protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
             [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
-            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            var endpoint = string.Empty;
+            if (Options.Fields.Count > 0) {
+                endpoint = $"{Options.UserInformationEndpoint}/~:({string.Join(",", Options.Fields)})";
```

Comment:
Instead of adding `~:({0})` to the UserInformationEndpoint, I am having it do a check for a `~` and insert the field after that if they are supplied. This eliminates the issue of a user clearing out all the `Fields` and the url ending up being invalid with a `~:()` at the end of it. A url of `https://api.linkedin.com/v1/people/~` is valid with LinkedIn.